File: 0033. Search in Rotated Sorted Array/solution.cpp
#include <vector>

class Solution {
public:
    int search(std::vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        int mid;

        while(l <= r){
            mid = (l + r) / 2;

            if (nums[mid] == target){
                return mid;
            }

            else if (nums[l] <= nums[mid]){
                if (target > nums[mid] || nums[l] > target){
                    l = mid + 1;
                }

                else{
                    r = mid - 1;
                }
            }

            else{
                if (target > nums[r] || target < nums[mid]){
                    r = mid - 1;
                }

                else{
                    l = mid + 1;
                }
            }
        }

        return -1;
    }
};

File: 0033. Search in Rotated Sorted Array/solution.py
class Solution(object):
    def search(self, nums, target,l=None,r=None):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if l is None: l = 0
        if r is None: r = len(nums) - 1
        if l > r: return -1

        mid = (l + r) // 2

        if nums[mid] == target:
            return mid

        left =  self.search(nums,target,l,mid -1)
        right = self.search(nums,target,mid+1,r)

        return max(left, right)
    